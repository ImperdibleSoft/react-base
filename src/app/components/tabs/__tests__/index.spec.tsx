import * as React from 'react';
import { shallow } from 'enzyme';

import Tabs from '..';
import Tab from '../components/tab';

const textTab = { id: 'test-text', label: 'Test' };
const linkTab = { id: 'test-link', label: 'Link', to: '/' };
const iconTab = { id: 'test-icon', icon: 'plus' };

describe('Tab', () => {
  it('should render a text tab', () => {
    const component = shallow(<Tab handleClick={() => {}} isActive={false} options={textTab} reference={{}} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a link tab', () => {
    const component = shallow(<Tab handleClick={() => {}} isActive={false} options={linkTab} reference={{}} />);
    expect(component).toMatchSnapshot();
  });

  it('should render an icon tab', () => {
    const component = shallow(<Tab handleClick={() => {}} isActive={false} options={iconTab} reference={{}} />);
    expect(component).toMatchSnapshot();
  });
});

describe('Tabs', () => {
  it('should render 2 tabs', () => {
    const component = shallow(<Tabs activeTab={'test-text'} options={[textTab, iconTab]} />);
    expect(component).toMatchSnapshot();
  });

  it('should render nothing', () => {
    const component = shallow(<Tabs activeTab={'test-text'} options={[]} />);
    expect(component).toMatchSnapshot();
  });
});
