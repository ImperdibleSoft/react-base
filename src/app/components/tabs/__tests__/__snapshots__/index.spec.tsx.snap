// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render a link tab 1`] = `
<StyledTab
  id="test-link"
  isActive={false}
  onClick={[Function]}
  role="button"
>
  <Link
    options={
      Object {
        "id": "test-link",
        "label": "Link",
        "to": "/",
      }
    }
    shouldInherit={true}
  />
</StyledTab>
`;

exports[`Tab should render a text tab 1`] = `
<StyledTab
  id="test-text"
  isActive={false}
  onClick={[Function]}
  role="button"
>
  <Button
    options={
      Object {
        "id": "test-text",
        "label": "Test",
        "type": "button",
      }
    }
    shouldInherit={true}
  />
</StyledTab>
`;

exports[`Tab should render an icon tab 1`] = `
<StyledTab
  id="test-icon"
  isActive={false}
  onClick={[Function]}
  role="button"
>
  <Button
    options={
      Object {
        "icon": "plus",
        "id": "test-icon",
        "type": "button",
      }
    }
    shouldInherit={true}
  />
</StyledTab>
`;

exports[`Tabs should render 2 tabs 1`] = `
<Wrapper>
  <StyledTabs
    minWidth={0}
  >
    <Tab
      handleClick={[Function]}
      isActive={true}
      key="test-text"
      options={
        Object {
          "id": "test-text",
          "label": "Test",
        }
      }
      reference={
        Object {
          "current": null,
        }
      }
    />
    <Tab
      handleClick={[Function]}
      isActive={false}
      key="test-icon"
      options={
        Object {
          "icon": "plus",
          "id": "test-icon",
        }
      }
      reference={
        Object {
          "current": null,
        }
      }
    />
  </StyledTabs>
</Wrapper>
`;

exports[`Tabs should render nothing 1`] = `""`;
